Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift

```diff
@@ -0,0 +1,144 @@
+
+    // MARK: - Init
+
+    init(repository: ReviewsRepository = ServiceLocator.provider.makeReviewsRepository()) {
+        self.repository = repository
+    }
+}
+
+// MARK: - ReviewProductViewModel
+
+extension ReviewProductViewModel: ReviewProductViewModelInput {
+
+    func getCellViewModel(at indexPath: IndexPath) -> ReviewerCellViewModel {
+        return cellViewModels[indexPath.row]
+    }
+}
+
+// MARK: - ReviewProductViewModelOutput
+
+extension ReviewProductViewModel: ReviewProductViewModelOutput {
+
+    func loadReviews() {
+        state = .loading
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+            self.repository.loadReviews { [weak self] reviews, error in
+                guard let self = self else { return }
+                guard let reviews = reviews else {
+                    self.state = .error
+                    self.alertMessage = error?.localizedDescription ?? " "
+                    return
+                }
+                self.processFetchedReviews(reviews: reviews)
+                self.navBarRating = reviews.rating
+                self.state = .loaded
+            }
+        }
+    }
+
+    func configureFetchRatingDetails(completion: @escaping (RatingDetailsViewModel?) -> Void) {
+        fetchRatingDetailsClosure = completion
+    }
+}
+
+// MARK: - Private Handlers
+
+private extension ReviewProductViewModel {
+
+    private func createCellViewModel(reviews: Domain.Review) -> ReviewerCellViewModel {
+        return ReviewerCellViewModel(reviewerImageUrl: reviews.image,
+                                     reviewerName: reviews.name,
+                                     reviewerComment: reviews.reviewDescription,
+                                     rating: Double(reviews.rating))
+    }
+
+    private func createRatingDetailsViewModel(reviews: Domain.Reviews) -> RatingDetailsViewModel {
+        return RatingDetailsViewModel(rating: reviews.rating,
+                                      fiveStarNumber: reviews.fiveStarRatingNumber,
+                                      fourStarNumber: reviews.fourStarRatingNumber,
+                                      threeStarNumber: reviews.threeStarRatingNumber,
+                                      twoStarNumber: reviews.twoStarRatingNumber,
+                                      oneStarNumber: reviews.oneStarRatingNumber,
+                                      totalReviewsNumber: reviews.totalRatingNumber)
+    }
+
+    private func processFetchedReviews(reviews: Domain.Reviews) {
+        reviewsList = reviews.reviews
+        var cellVMs = [ReviewerCellViewModel]()
+        for review in reviewsList {
+            cellVMs.append(createCellViewModel(reviews: review))
+        }
+        cellViewModels = cellVMs
+
+        let ratingDetailsView: RatingDetailsViewModel = createRatingDetailsViewModel(reviews: reviews)
+        ratingDetails = ratingDetailsView
+    }
+}
+
+// MARK: - states
+
+enum State {
```

Comment:
This is a specific view state, let's add it to `ReviewProductViewModelType.swift` and rename it. 
```suggestion
enum ReviewProductState {
```